src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -628,7 +628,27 @@ internal override void Dispose(bool isDisposing)
             {
                 _cmdTransportManagers.Clear();
                 _closeTimeOutTimer.Dispose();
+
+                // Stop session processing thread.
+                try
+                {
+                    _sessionMessageQueue.CompleteAdding();
+                }
+                catch (ObjectDisposedException)
+                {
+                    // Object already disposed.
+                }
                 _sessionMessageQueue.Dispose();
+
+                // Stop command processing thread.
+                try
+                {
+                    _commandMessageQueue.CompleteAdding();
+                }
+                catch (ObjectDisposedException)
+                {
+                    // Object already disposed.
+                }
```

Comment:
CodeFactor: Statements or elements wrapped in curly brackets must be followed by a blank line.
```suggestion
                }

```